test/topic/ReliableTopicTest.js

```diff
@@ -130,97 +121,73 @@ describe('TopicTest', function () {
+            topic.addMessageListener(function (m) {
+                if (m.messageObject === 20) {
                     done();
                 }
             });
             const all = generateItems(clientOne, 20);
             ringbuffer.addAll(all);
-        });
+        }).catch(done);
     });
 
-    it('discards the item when there is no more space', function () {
-        let topic;
-        let ringbuffer;
-        return clientOne.getReliableTopic('discard').then(function (t) {
-            topic = t;
-            return topic.getRingbuffer();
-        }).then(function (rb) {
-            ringbuffer = rb;
-            const all = generateItems(clientOne, 10);
-            ringbuffer.addAll(all);
-            return topic.publish(11);
-        }).then(function () {
-            return ringbuffer.tailSequence();
-        }).then(function (seq) {
-            return ringbuffer.readOne(seq);
-        }).then(function (item) {
-            const obj = clientOne.getSerializationService().toObject(item.payload);
-            expect(obj).to.equal(10);
-        });
+    it('discards the item when there is no more space', async function () {
+        const topic = await clientOne.getReliableTopic('discard');
+        const ringbuffer = topic.getRingbuffer();
+
+        const all = generateItems(clientOne, 10);
+        await ringbuffer.addAll(all);
+        await topic.publish(11);
+
+        const seq = await ringbuffer.tailSequence();
+        const item = await ringbuffer.readOne(seq);
+        const obj = clientOne.getSerializationService().toObject(item.payload);
+        expect(obj).to.equal(10);
     });
 
-    it('overwrites the oldest item when there is no more space', function () {
-        let topic;
-        let ringbuffer;
-        return clientOne.getReliableTopic('overwrite').then(function (t) {
-            topic = t;
-            return topic.getRingbuffer();
-        }).then(function (rb) {
-            ringbuffer = rb;
-            const all = generateItems(clientOne, 10);
-            ringbuffer.addAll(all);
-            return topic.publish(11);
-        }).then(function () {
-            return ringbuffer.tailSequence();
-        }).then(function (seq) {
-            return ringbuffer.readOne(seq);
-        }).then(function (item) {
-            const obj = clientOne.getSerializationService().toObject(item.payload);
-            expect(obj).to.equal(11);
-        });
+    it('overwrites the oldest item when there is no more space', async function () {
+        const topic = await clientOne.getReliableTopic('overwrite');
+        const ringbuffer =  topic.getRingbuffer();
+
+        const all = generateItems(clientOne, 10);
+        await ringbuffer.addAll(all);
+        await topic.publish(11);
+
+        const seq = await ringbuffer.tailSequence();
+        const item = await ringbuffer.readOne(seq);
+        const obj = clientOne.getSerializationService().toObject(item.payload);
+        expect(obj).to.equal(11);
```

Comment:
Not related to this PR, but do we have an issue about migrating tests to `async/await`? It is a lot readable :smile:

Reply:
Yeah, it's a lot more readable. I was going to migrate tests incrementally as a part of other PRs. But having this plan defined as a GH issue is a good idea. Going to create one.

Reply:
Here it goes: #569